File: app/src/main/java/tn/esprit/wayfinder/models/ChatModels.kt
package tn.esprit.wayfinder.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class ChatModel {
    @SerialName("huggingface")
    HUGGINGFACE,
    
    @SerialName("openai_gpt4o_mini")
    OPENAI_GPT4O_MINI,
    
    @SerialName("openai_gpt4o")
    OPENAI_GPT4O
}

@Serializable
data class ChatMessageRequest(
    val message: String,
    val model: ChatModel? = null
)

@Serializable
data class FlightPack(
    val title: String,
    val price: String,
    val origin: String,
    val destination: String,
    val airline: String? = null,
    val details: String? = null
)

@Serializable
data class ChatMessageResponse(
    val message: String,
    @SerialName("model_used") val modelUsed: String,
    @SerialName("flight_packs") val flightPacks: List<FlightPack>? = null,
    @SerialName("session_id") val sessionId: String
)

@Serializable
data class SwitchModelRequest(
    val model: ChatModel
)

@Serializable
data class SwitchModelResponse(
    val success: Boolean,
    val model: String
)

@Serializable
data class ChatHistoryItem(
    val message: String,
    val role: String, // "user" or "assistant"
    @SerialName("model_used") val modelUsed: String,
    @SerialName("flight_packs") val flightPacks: List<FlightPack>? = null,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class ClearHistoryResponse(
    val success: Boolean
)

@Serializable
data class AvailableModel(
    val id: String,
    val name: String,
    val available: Boolean
)

@Serializable
data class AvailableModelsResponse(
    val models: List<AvailableModel>
)

File: WayFinder/Models/ChatModels.swift
import Foundation

enum ChatModel: String, Codable, CaseIterable {
    case huggingface = "huggingface"
    case openaiGpt4oMini = "openai_gpt4o_mini"
    case openaiGpt4o = "openai_gpt4o"
    
    var displayName: String {
        switch self {
        case .huggingface:
            return "Hugging Face (Free)"
        case .openaiGpt4oMini:
            return "OpenAI GPT-4o Mini"
        case .openaiGpt4o:
            return "OpenAI GPT-4o"
        }
    }
}

struct ChatMessageRequest: Encodable {
    let message: String
    let model: ChatModel?
    
    enum CodingKeys: String, CodingKey {
        case message
        case model
    }
}

struct FlightPack: Decodable {
    let title: String
    let price: String
    let origin: String
    let destination: String
    let airline: String?
    let details: String?
}

struct ChatMessageResponse: Decodable {
    let message: String
    let modelUsed: String
    let flightPacks: [FlightPack]?
    let sessionId: String
    
    enum CodingKeys: String, CodingKey {
        case message
        case modelUsed = "model_used"
        case flightPacks = "flight_packs"
        case sessionId = "session_id"
    }
}

struct SwitchModelRequest: Encodable {
    let model: ChatModel
}

struct SwitchModelResponse: Decodable {
    let success: Bool
    let model: String
}

struct ChatHistoryItem: Decodable {
    let message: String
    let role: String // "user" or "assistant"
    let modelUsed: String
    let flightPacks: [FlightPack]?
    let createdAt: String?
    
    enum CodingKeys: String, CodingKey {
        case message
        case role
        case modelUsed = "model_used"
        case flightPacks = "flight_packs"
        case createdAt = "created_at"
    }
}

struct ClearHistoryResponse: Decodable {
    let success: Bool
}

struct AvailableModel: Decodable {
    let id: String
    let name: String
    let available: Bool
}

struct AvailableModelsResponse: Decodable {
    let models: [AvailableModel]
}

